### PR TITLE
refactor: Improve pattern handling and enhance test coverage for `Naming` class.

### PR DESCRIPTION
**Pull request by terabytesoftw**
# Pull Request

| Q            | A                                                                  |
| ------------ | ------------------------------------------------------------------ |
| Is bugfix?   | ✔️                                                              |
| New feature? | ❌                                                              |
| Breaks BC?   | ❌                                                              |

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved pattern handling and character normalization in naming utilities.
* **Tests**
  * Added comprehensive test coverage for input ID generation with multibyte character support.
  * Added tests for naming utility functions and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/Base/BaseNaming.php [hunk 1]
```diff
@@ -73,22 +73,15 @@ public static function convertToPattern(string $regexp, string|null $delimiter =
...
+        $endPosition = strrpos((string) $pattern, $delimiter, 1);
 
         if ($endPosition === false) {
             throw new InvalidArgumentException(
                 Message::INCORRECT_REGEXP->getMessage(),
             );
         }
 
-        return substr($pattern, 1, $endPosition - 1);
+        return substr((string) $pattern, 1, $endPosition - 1);
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/Base/BaseNaming.php]: Crea un tests donde falle, y hare el cambio.

[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/Base/BaseNaming.php]: There was 1 failure:

1) UIAwesome\Html\Helper\Tests\NamingTest::testConvertToPatternWithMalformedUtf8
Failed asserting that exception of type "InvalidArgumentException" is thrown.

FAILURES!
Tests: 228, Assertions: 240, Failures: 1.

There was 1 failure:

1) UIAwesome\Html\Helper\Tests\NamingTest::testConvertToPatternWithPregReplaceFailure
Failed asserting that exception of type "ValueError" matches expected exception "InvalidArgumentException". Message was: "strrpos(): Argument 3 ($offset) must be contained in argument 1 ($haystack)" at
/home/terabytesoftw/Github/ui-awesome/html-helper/src/Base/BaseNaming.php:76
/home/terabytesoftw/Github/ui-awesome/html-helper/tests/NamingTest.php:185

[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/Base/BaseNaming.php]: There was 1 failure:

1) UIAwesome\Html\Helper\Tests\NamingTest::testConvertToPatternWithPregReplaceFailure
Failed asserting that exception of type "ValueError" matches expected exception "InvalidArgumentException". Message was: "strrpos(): Argument 3 ($offset) must be contained in argument 1 ($haystack)" at
/home/terabytesoftw/Github/ui-awesome/html-helper/src/Base/BaseNaming.php:76
/home/terabytesoftw/Github/ui-awesome/html-helper/tests/NamingTest.php:182
